src/type/scalars.js

```diff
@@ -31,6 +31,15 @@ function serializeInt(outputValue: mixed): number {
     num = Number(outputValue);
   }
 
+  if (
+    outputValue !== null &&
+    typeof outputValue === 'object' &&
+    typeof outputValue.valueOf === 'function' &&
+    !Array.isArray(outputValue)
+  ) {
+    num = Number(outputValue.valueOf());
+  }
```

Comment:
Please use `serializeObject` instead and add it `Int`, `Float` and `Boolean`.
Also please update tests in https://github.com/graphql/graphql-js/blob/master/src/type/__tests__/serialization-test.js

So basically, please replicate how this is done for ID and String.

Reply:
Don't you think than using of serializeObject method is incorrect in this situation (it contains checking that object has toJSON() method

Reply:
@IvanGoncharov  I did it

Reply:
> Is it good idea to squash my 4 commits to one?

@alex-knyazev Both work, since it's small commit I can just "squash and merge" in GitHub UI.
Please don't forget to make the same changes and tests for `GraphQLBoolean` type.

Reply:
@IvanGoncharov  I've added boolean type